### PR TITLE
Adds the ability to get a Recognizer from a Manger using the base name or and the value from event.type

### LINKED ISSUE #745: Manager.get(<string>) Should Return Recognizers from Event Names
In [Angular Hammer](https://github.com/RyanMullins/angular-hammer) I try to use the `event.type` property (which is the name of the event) to [get a recognizer from a Manager instance](https://github.com/RyanMullins/angular-hammer/blob/master/angular.hammer.js#L145). Unfortunately, for events like `panmove`, `pressup`, and so on the Manger returns null, which caused a mild problem for me, though it was easily fixed. 

When I looked into this, I found that [Hammer Managers `get()` function](https://github.com/hammerjs/hammer.js/blob/master/src/manager.js#L126) takes in either a recognizer or a string argument, and returns either that same [recognizer](https://github.com/hammerjs/hammer.js/blob/master/src/manager.js#L128) or the [recognizer that responds to that event name](https://github.com/hammerjs/hammer.js/blob/master/src/manager.js#L134). In the documentation [you state](http://hammerjs.github.io/api/#get(string),-add(recognizer)-and-remove(recognizer)): 

> The get and remove methods takes the event name (from a recognizer) or a recognizer instance as an argument.

This implementation seems limited to me. 

The only situation where it would be useful is if you had one Recognizer available in a scope such that the type of that Recognizer is unknown, and want to instantiate a new Recognizer of the same type for a different Manager inside a function defined within that scope.

I would like to see this function's implementation expanded so that I can pass in the value of `event.type` and have it spit out a Recognizer instance. 

I am working on this feature in fork and will submit my implementation as a pull request. Please let me know if there is specific implementation concept you had in mind for a similar extension.

### PR DESCRIPTION
**Pull request by RyanMullins**
If accepted, this would resolve #745

### CODE REVIEW

**Comment by runspired** [comment]
@arschmitz this represents a departure in intentional behavior.  I like it, but it should be at a min a minor version bump and a warning.

File: hammer.js [hunk 1]
```diff
@@ -320,8 +320,8 @@ function uniqueId() {
  * @returns {DocumentView|Window}
  */
 function getWindowForElement(element) {
-    var doc = element.ownerDocument;
-    return (doc.defaultView || doc.parentWindow);
+    var doc = element.ownerDocument || element;
+    return (doc.defaultView || doc.parentWindow || window);
```
Comment (runspired) [inline hammer.js]: This should be discussed in context of #598 
I also don't like that we're adding unrelated changes here.

File: hammer.js [hunk 2]
```diff
@@ -1114,7 +1114,7 @@ TouchAction.prototype = {
             value = this.compute();
         }
 
-        if (NATIVE_TOUCH_ACTION) {
+        if (NATIVE_TOUCH_ACTION && this.manager.element.style) {
```
Comment (runspired) [inline hammer.js]: @RyanMullins what was the reasoning here?

**Comment by runspired** [comment]
This needs cleaned up, changes not related to the feature at hand need to be removed or explained.

This is definitely a candidate feature for minor release.

**Comment by arschmitz** [comment]
@RyanMullins do you plan on coming back to this?

**Comment by RyanMullins** [comment]
Unfortunately I'm unable to see a point in time when I will be able to come back to this.
